ui/src/tool-panel/location-track/dialog/location-track-ratko-push-dialog.tsx

```diff
@@ -65,107 +66,84 @@ export const LocationTrackRatkoPushDialog: React.FC<LocationTrackRatkoPushDialog
     const [startKm, setStartKm] = React.useState<KmNumber>();
     const [endKm, setEndKm] = React.useState<KmNumber>();
     const kmOptions = startAndEndPoints && getKmOptions(startAndEndPoints);
-    const [pushing, setPushing] = React.useState(false);
-    const canPush = startKm && endKm && !pushing;
-    const [pushDone, setPushDone] = React.useState(false);
+    const canPush = !!startKm && !!endKm;
 
     async function pushToRatko() {
-        if (locationTrack && startKm && endKm && !pushing) {
+        if (locationTrack && startKm && endKm) {
             try {
-                setPushing(true);
                 const kms = getKmsInRange(Number.parseInt(startKm), Number.parseInt(endKm));
 
-                await pushLocationTracksToRatko([
+                pushLocationTracksToRatko([
                     {
                         locationTrackId: locationTrack.id,
                         changedKmNumbers: kms,
                     },
                 ]);
-
-                setPushDone(true);
+                Snackbar.success(
+                    t('tool-panel.location-track.ratko-push-dialog.pushing-started'),
+                    t('tool-panel.location-track.ratko-push-dialog.follow-in-ratko'),
+                );
             } finally {
-                setPushing(false);
+                props.onClose();
```

Comment:
Pullarin effektiivinen ja hyvin ruma pihvi täällä. Sen sijasta että jäätäisiin odottamaan pyynnön palaamista, näytetään vaan toasti siirron käynnistymisestä jossa pyydetään operaattoria seuraamaan raidetta Ratkossa. Hyvin rumana sivuvaikutuksena on tosin se, että nythän täällä ei sitten pystytä tarkastelemaan mitään tuohon kutsuun liittyen - se vaan ammutaan menemään ja sen jälkeen unohdetaan.

Tämän fiksuunnuttaminen käytännössä vaatisi sen, että viennin saisi bäkkärin sisäisesti asynkroniseksi. En oikein keksinyt nopeasti mitään yhtä aikaa helppoa ja ei-typerää tapaa tehdä tästä sellaista (voisihan tuolle tehdä bäkkäri-`Task`:in kantatauluineen ja scheduleineen, mutta tuollaisen isomman refakin ei katsottu maksavan vaivaa). Ajattelin sitten että lienee parempi että kaikki tästä aiheutuva extrarumuus on sitten frontin päässä. Varmaan bäkkärille voisi jonkun uuden threadin spawnata ja olla jäämättä odottelemaan sen palaamista ennen frontille vastauksen välittämistä, mutta sitten kaikki bäkkärin thread-contextit jne. menisi jänniksi sillä spawnatulla threadilla, ja ruma sekin...

Jos joku keksii parempia vaihtoehtoja, niin otetaan vastaan.

Reply:
Joo tämä on aika jännä kyllä, menee rakenteet vähän solmuun. Mutta kuten todettu, sen verran spesiaaliominaisuus ettei kannata lähteä tekemään kovin hienoja rakenteita tätä varten.